server/src/main/java/org/opensearch/index/search/QueryStringQueryParser.java

```diff
@@ -374,6 +383,13 @@ public Query getFieldQuery(String field, String queryText, boolean quoted) throw
         }
     }
 
+    private void onDiscoveredField(String field) {
```

Comment:
Minor, I would rename it to `addDiscoveredQueryField`

Reply:
done